Move all documentation out of modals and into Help sidebar or inline

esthersweon opened the pull request:
**User-Facing Changes**
- Move desktop app's Help menu items to the Help sidebar
   <img width="308" alt="Screen Shot 2021-11-24 at 12 25 26 AM" src="https://user-images.githubusercontent.com/6993359/143201579-d943cfd6-d01f-4561-afcc-219d4b7b9768.png">
   <img width="362" alt="Screen Shot 2021-11-24 at 12 21 02 AM" src="https://user-images.githubusercontent.com/6993359/143201175-b918b10b-cfea-46a4-a230-c53d1c2c074f.png">

- Add desktop app's Help menu item to open the Help sidebar
   <img width="217" alt="Screen Shot 2021-11-24 at 12 24 24 AM" src="https://user-images.githubusercontent.com/6993359/143201454-23fa1496-09b2-40c3-8033-9e1db905fb13.png">

- Inline all docs for sidebar tabs
   <img width="433" alt="Screen Shot 2021-11-24 at 7 53 43 PM" src="https://user-images.githubusercontent.com/6993359/143377064-601834dd-b891-4b98-8fbc-17c980999758.png">
   <img width="441" alt="Screen Shot 2021-11-24 at 7 53 39 PM" src="https://user-images.githubusercontent.com/6993359/143377068-d836ebc5-0724-4346-aac0-ba78c2f58e79.png">

- Make internal links to message path syntax docs work properly

<!-- link relevant github issues -->
- Resolves https://github.com/foxglove/studio/issues/2163
<!-- add `docs` label if this PR requires documentation updates -->

2metres commented:
- Do we have to hide this content?
- Can we change "Learn more" to something like "Jump to docs" or similar
- Remove the italicized font

esthersweon commented:
Implemented team's feedback.

Assigning to @defunctzombie for final code review. Did a bunch of refactors on how I implemented the feature; would love feedback.

defunctzombie commented on packages/studio-base/src/panels/Table/index.help.md:
```diff
@@ -2,4 +2,4 @@ Display topic data in a tabular format.
 Use [message path syntax](#help:message-path-syntax) to determine the data to display. Click a column's header to sort by that column; hold `Shift` to click multiple columns.
 
-[Learn more](https://foxglove.dev/docs/panels/table).
+[Jump to docs](https://foxglove.dev/docs/panels/table).
```
Jump to docs sounds strange. `Open docs`? The word jump feels a-typical especially here since it will open a browser window.

esthersweon replied:
"View docs"?

defunctzombie replied:
👍
